File: api/app/interfaces/schemas/briefing.py
"""主动情报简报接口 schema。"""

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel

from app.domain.models.intel_briefing import IntelBriefing


class BriefingItemSchema(BaseModel):
    """一条情报要点的对外表示。"""
    title: str
    category: str
    reason: str
    action: str
    urgency: str


class BriefingResponse(BaseModel):
    """情报简报响应（latest 为 None 时返回 has_briefing=false）。"""
    has_briefing: bool
    headline: str = ""
    items: List[BriefingItemSchema] = []
    generated_by: str = ""
    disclaimer: str = ""
    generated_at: Optional[datetime] = None

    @classmethod
    def from_domain(cls, briefing: Optional[IntelBriefing]) -> "BriefingResponse":
        if briefing is None:
            return cls(has_briefing=False)
        return cls(
            has_briefing=True,
            headline=briefing.headline,
            items=[
                BriefingItemSchema(
                    title=i.title, category=i.category, reason=i.reason,
                    action=i.action, urgency=i.urgency.value,
                )
                for i in briefing.items
            ],
            generated_by=briefing.generated_by,
            disclaimer=briefing.disclaimer,
            generated_at=briefing.generated_at,
        )

File: ui/src/lib/api/briefing.ts
import { get, post } from "./fetch";

// ==================== 主动情报简报类型 ====================

/** 情报项紧迫度 */
export type BriefingUrgency = "high" | "normal" | "low";

/** 一条情报要点 */
export type BriefingItem = {
  title: string;
  category: string;
  reason: string;
  action: string;
  urgency: BriefingUrgency;
};

/** 情报简报 */
export type Briefing = {
  has_briefing: boolean;
  headline: string;
  items: BriefingItem[];
  generated_by: string; // llm | fallback
  disclaimer: string;
  generated_at: string | null;
};

// ==================== 主动情报简报 API ====================

export const briefingApi = {
  /** 获取最新一份情报简报（从未生成则 has_briefing=false） */
  latest: (): Promise<Briefing> => {
    return get<Briefing>("/briefings/latest");
  },

  /** 立即生成一份情报简报 */
  generate: (): Promise<Briefing> => {
    return post<Briefing>("/briefings/generate");
  },
};
